src/geometry/CTiglRectGridSurface.h

```diff
@@ -0,0 +1,331 @@
+/*
+* Copyright (C) 2021 German Aerospace Center (DLR/SC)
+*
+* Created: 2021-01-21 Jan Kleinert <Jan.Kleinert@dlr.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef CTIGLRECTGRIDSURFACE_H
+#define CTIGLRECTGRIDSURFACE_H
+
+#include <type_traits>
+#include <vector>
+
+#include "tigl_internal.h"
+#include "CTiglError.h"
+
+#include <TopoDS_Shape.hxx>
+#include <TopoDS_Face.hxx>
+#include <gp_Pnt.hxx>
+#include <TopTools_IndexedMapOfShape.hxx>
+#include <TopExp.hxx>
+#include <TopoDS.hxx>
+#include <BRepTools.hxx>
+#include <Geom_Surface.hxx>
+#include <BRep_Tool.hxx>
+
+namespace {
+struct DefaultAnnotation
+{};
+}
+
+
+namespace tigl
+{
+
+/**
+ * @brief The CTiglRectGridSurface class represents
+ * a collection of faces arranged in a rectangular
+ * grid. Optionally, the faces can be annotated using
+ * a custom type.
+ */
+template <typename Annotation = DefaultAnnotation>
+class CTiglRectGridSurface
```

Comment:
Great solution. I really like, that you allow to annotate / attach anything to the faces. It's similar to the metadata idea I have for topologies in general.

By the way, I created a class which is similar but not entirely for face strips, which is much less fancy. Maybe, we should create a new issue to integrate `CTiglCompoundSurface` into your class. 

What my class adds to yours is the functionality of quering points from the shape compound. Conceptually, `CTiglRectGridSurface` operates on shapes, `CTiglCompoundSurface` on surfaces though. For now, nothing has to be done.

Reply:
See #776